sm/tedge_agent/src/agent.rs

```diff
@@ -113,15 +116,19 @@ pub struct SmAgent {
     config: SmAgentConfig,
     name: String,
     persistance_store: AgentStateRepository,
+    _flock: Flockfile,
 }
 
 impl SmAgent {
     pub fn new(name: &str, config: SmAgentConfig) -> Self {
         let persistance_store = AgentStateRepository::new(config.sm_home.clone());
+        let flock = check_another_instance_is_not_running(&name).unwrap();
```

Comment:
This `unwrap()` is not good, it causes a panic when launched the second agent.
```
2021-09-20T08:38:00.986+00:00 ERROR flockfile::unix: Another instance of tedge_agent is running.
2021-09-20T08:38:00.986+00:00 ERROR flockfile::unix: Lock file path: /run/lock/tedge_agent.lock
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: FromNix { path: "/run/lock/tedge_agent.lock", source: Sys(EAGAIN) }', sm/tedge_agent/src/agent.rs:125:66
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
In mapper, the function which calls `check_another_instance_is_not_running()` returns `Result`. One option is change the return type to `Result<Self, FlockfileError>`, then 
```suggestion
        let flock = check_another_instance_is_not_running(&name)?;
```
I recommend you to talk with @makr11st.